frontend/src/bundles/profile/components/profile-settings/profile-settings.tsx

```diff
@@ -147,16 +149,24 @@ const ProfileSettings: React.FC<Properties> = ({
                     fullName: (data.fullName || '').trim(),
                     username: data.username ? data.username.trim() : null,
                 };
+
                 onSubmit(payload);
             })(event_);
         },
-        [handleSubmit, onSubmit],
+        [handleSubmit, onSubmit, getValues],
     );
 
     const closeModal = useCallback((): void => {
         setIsOpen(false);
     }, []);
 
+    const handleCheckboxChange = useCallback(
+        (event: React.ChangeEvent<HTMLInputElement>): void => {
+            setValue('isPublic', event.target.checked);
+        },
+        [setValue],
+    );
+
```

Comment:
Do we need this callback for handling the checkbox? I guess, it is possible to handle it inside the checkbox component and set the value for a field

Reply:
I tried to find a way to control the checkbox value without the handler, but it was unsuccessful for me.
I tried the variant with {..register} but it didn't work.
Maybe if you have some suggestions, I would be really thankful.

Reply:
I pushed suggestion)